File: read.js
import * as jose from 'https://deno.land/x/jose@v5.4.1/index.ts'
import private_jwk from './keys/privateKey.json' with { type: 'json' }

async function decryptJWE(jwe) {
	try {
		const privateKey = await jose.importJWK(private_jwk)
		const {
			plaintext,
			// protectedHeader,
		} = await jose.compactDecrypt(jwe, privateKey)

		// console.log(protectedHeader)
		return new TextDecoder().decode(plaintext)
	} catch (error) {
		console.error('Failed to decrypt JWE:', error);
	}
}

const jwe = 'eyJhbGciOiJSU0EtT0FFUC0yNTYiLCJlbmMiOiJBMjU2R0NNIn0.sguLZCnlUVcZFsjwrczOjJ3KMV8y05o_IpDiTutyycSZf__S5JSXaasA9QWhO6kwAuTs4SlLyA0A5nJG_fJ55xXbhxByrzEszhK6UyixyvMUEo-PGBLjySDoWNZVRN1bG0K88M5AMV2OeBhKGhRkN4TWwtFEkNWz37r3r7YEHX3yNYZ6GbB43xeySp1V3p0u_kCfPXrBqdCSUslEy3lN-4IlqxGa-qiHw9Lxq3Y5mfC8Hm7pfQIt-3tTT-p-JRDzgI8bhiXoCF4hdILn3WJayPEGlX81b9MvG5Jo7EZPfayVxsae6aj1keH6l5UPxeW0ylD8f5L35nm2a7RBZPRG2_cGcY1zOJIox8F-lhJxzHcCnVy1ZskwaNSKhws5kTqq-k5MP7QI1dlf5znK-w3e9pmGCekCa9RHfAYW5DTXGbB_pD3TmacwVRrz0-_FoKreDtevkbk56D_tkae6u36dH7fvY7K93m_1_nc3OgPCiCHKNIESPk0rODE_yBUv8eUg0Ni-oQ8BIKus4Uo3NbwZxBEnTixdiu2jsKPEmcOX5ZVHv9uGRCk9Ok1YykFvPxOEmbtR_t0uJ4L_wIf5Ysybkch90lV87dLH8l-0-eqmBKtcinhiP97pZ7mI50zOBAiouDVSJ29-M28Jo3JlUg4BnPvzF3t1F6BBaQNAQghjJ58.rNdxdIT47pcoIHmG.VCNnuSoSz6SjYl1VSXE9ZQHb7EpnHCYkjwm8wA-EXG2mKq_vTaAAA3CBz6lUUerbsDJ-k76TV_tLaMwhBUEdcDBCJbyfL0MR8kAkDvf75VCJUo7iLwvDk9GNnyswrTUAM2g_IxlMbuPDPQ.88q0RtL7M1srvtXwaTxCLA';
const data = await decryptJWE(jwe)
console.log('Decrypted payload:', data)

File: read.py
import json
from authlib.jose import JsonWebEncryption
from authlib.jose.errors import JoseError

def decrypt_jwe(jwe_token, jwk_file):
    with open(jwk_file) as f:
        private_jwk = json.load(f)

    # If 'key_ops' is not required, ensure this is removed if it causes compatibility issues
    private_jwk.pop('key_ops', None)

    jwe_cryptor = JsonWebEncryption()

    try:
        decrypted_payload = jwe_cryptor.deserialize_compact(jwe_token, private_jwk)
        return decrypted_payload['payload'].decode('utf-8')
    except JoseError as e:
        print("Failed to decrypt JWE:", e)

# Example usage
jwe = 'eyJhbGciOiJSU0EtT0FFUC0yNTYiLCJlbmMiOiJBMjU2R0NNIn0.Z2CAMaH_FnXhlBN76naa-Pu17_M7wrqilQJVTqkbDnDk5hKxaPdYFBbUVb2rgY2yfVxU8Bou8dLqZjHm03WFGQslPUotSgAyq4TbUdbiZjzjHB8QSQFRiko1t7f18jWpVnOdiYzhk2jkLKzvywRbQ2y5Y27f35a6olR0TBuHlNUUnx5GujAu4b6_zT0Ru9p6jdh3lZw1SS97rzy5ouTvcJxLGvuQBRP6PxWbNbOU9NJkcGz65v_oob0AQe6svn7K_yQIsAGHRCI-sPIZQ1ZWj23ydvq7VyCrhmJXIcrvLU7zT1o-OD0P4Lhzewf5NuR0ZS3z551Uk7BFCA85QJB1OXLq6Ruuef6QUGvRKFKuRtiunKlGAbZLO5jg-9bHmeSi5qy6tTF8xLFqVbqth1Tbs92rS8WktDswVl9OzmY3KccrNitKKxD5D5sDqok9wZ7AX_stJFG2iQ8z_XbPIs036m-qjF0pBUa0fzdXwUQ_z82BtoX4MjVsBstP7qkXgvx8vokht9VK0m1m76zoXRtOVLkTdoHJXYqd7wtYTJHBaYrsQz3xpQVD3LOWIoCdJs_R3DR5VkcwMIs2sDEEHkk6xEMMI8QmNbrANWnAqFlYKm9OTIupcDDbbEzzXsjVj7KIX22K8P5GjKW0SJQNzh9t0g2TAcDNZIVWtpVyss6sqBo.AYtH87yesZxDlNsO.Y5kplKSjGKiKg4CHSVH7gFO7Hc8zTkSDtDyIHZrBzVsQaNbDmj38eWgT32MAbH-yIeKPmBr9hIrv9UL45b7gqhZtKIYV2ebrjtlWNYdV_wmKsJ1NAB_Cmqy61gsjhK9HC-rVivBQEDZa3gGhzbeMyXwD7eiGdJkmCiPISqlCfr_29g47ftljGZduv5f47iZwYEbORDbb2X77tzvWJsrBrEcFkJYjesqk_vf-0Mb4KDp3bUq2yAPfXOPokWbmfycLPden7uUO2d484-NaRZLwkeem0grX.SuFG3SkE9fLzv1kYgtuWJw'
data = decrypt_jwe(jwe, './keys/privateKey.json')
print("Decrypted payload:", data)
